codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-batchmanager-sync-class.java

```diff
@@ -0,0 +1,88 @@
+import software.amazon.awssdk.protocols.json.JsonOperationMetadata;
+import software.amazon.awssdk.services.batchmanager.model.BatchManagerException;
+import software.amazon.awssdk.services.sqs.batchmanager.SqsBatchManager;
+import software.amazon.awssdk.utils.Logger;
+
+/**
+ * Internal implementation of {@link BatchManagerClient}.
+ *
+ * @see BatchManagerClient#builder()
+ */
+@Generated("software.amazon.awssdk:codegen")
+@SdkInternalApi
+final class DefaultBatchManagerClient implements BatchManagerClient {
+    private static final Logger log = Logger.loggerFor(DefaultBatchManagerClient.class);
+
+    private final SyncClientHandler clientHandler;
+
+    private final AwsJsonProtocolFactory protocolFactory;
+
+    private final SdkClientConfiguration clientConfiguration;
+
+    private final ScheduledExecutorService executorService;
+
+    protected DefaultBatchManagerClient(SdkClientConfiguration clientConfiguration) {
+        this.clientHandler = new AwsSyncClientHandler(clientConfiguration);
+        this.clientConfiguration = clientConfiguration;
+        this.protocolFactory = init(AwsJsonProtocolFactory.builder()).build();
+        this.executorService = clientConfiguration.option(SdkClientOption.SCHEDULED_EXECUTOR_SERVICE);
+    }
+
+    @Override
+    public final String serviceName() {
+        return SERVICE_NAME;
+    }
+
+    private static List<MetricPublisher> resolveMetricPublishers(SdkClientConfiguration clientConfiguration,
+                                                                 RequestOverrideConfiguration requestOverrideConfiguration) {
+        List<MetricPublisher> publishers = null;
+        if (requestOverrideConfiguration != null) {
+            publishers = requestOverrideConfiguration.metricPublishers();
+        }
+        if (publishers == null || publishers.isEmpty()) {
+            publishers = clientConfiguration.option(SdkClientOption.METRIC_PUBLISHERS);
+        }
+        if (publishers == null) {
+            publishers = Collections.emptyList();
+        }
+        return publishers;
+    }
+
+    private HttpResponseHandler<AwsServiceException> createErrorResponseHandler(BaseAwsJsonProtocolFactory protocolFactory,
+                                                                                JsonOperationMetadata operationMetadata) {
+        return protocolFactory.createErrorResponseHandler(operationMetadata);
+    }
+
+    private <T extends BaseAwsJsonProtocolFactory.Builder<T>> T init(T builder) {
+        return builder.clientConfiguration(clientConfiguration).defaultServiceExceptionSupplier(BatchManagerException::builder)
+                      .protocol(AwsJsonProtocol.REST_JSON).protocolVersion("1.1");
+    }
+
+    @Override
+    public void close() {
+        clientHandler.close();
+    }
+
+    @Override
+    public SqsBatchManager batchManager() {
+        return SqsBatchManager.builder().client(this).executor(executorService).scheduledExecutor(executorService).build();
```

Comment:
I'm not sure if using the scheduled executor service to send blocking request is a good idea. I was initially thinking of using the existing async response completion pool, but looks like that only gets set in async client,  so we should create a default executor service in each sync batch manager and close it when the batch manager gets closed.

Reply:
Yeah I noticed that too, that's why I switched to using the scheduled executor service. Will modify to create a default executor service.

Reply:
Currently creating a single thread executor inside sync batch manager and closing it when the client is closed. Not sure if single thread should be the expected behavior, could also make a executor with a maxPoolSize equal to the client's maxConnections?